aten/src/ATen/Dispatch.h

```diff
@@ -193,20 +185,19 @@ inline at::ScalarType scalar_type(at::ScalarType s) {
 
 #define AT_DISPATCH_SWITCH(TYPE, NAME, ...)                                 \
   [&] {                                                                     \
-    const auto& the_type = TYPE;                                            \
+    const at::ScalarType the_type = TYPE;                                   \
     constexpr const char* at_dispatch_name = NAME;                          \
     /* don't use TYPE again in case it is an expensive or side-effect op */ \
```

Comment:
This comment is out of place here now, why instead of keeping definition of `the_type` not say `at::ScalarType _st = TYPE;` and this will make this patch even smaller